pkg/ruler/compat.go

```diff
@@ -107,7 +112,7 @@ func MemstoreTenantManager(
 		memStore := NewMemStore(userID, queryFunc, metrics, 5*time.Minute, log.With(logger, "subcomponent", "MemStore"))
 
 		mgr := rules.NewManager(&rules.ManagerOptions{
-			Appendable:      NoopAppender{},
+			Appendable:      newAppendable(cfg, overrides, logger, userID),
```

Comment:
If the remote write is disable you could log an info and set NoopAppender.

then you can remove validation on enable.